test/functional/lua/ui_event_spec.lua

```diff
@@ -306,6 +306,92 @@ describe('vim.ui_attach', function()
       },
     })
   end)
+
+  it('ext_cmdline completion popupmenu', function()
+    screen:try_resize(screen._width, 10)
+    screen:add_extra_attr_ids { [100] = { background = Screen.colors.Black } }
+    exec_lua([[
+      vim.o.wildoptions = 'pum'
+      local buf = vim.api.nvim_create_buf(false, true)
+      vim.cmd('call setline(1, range(1, 10))')
+      _G.win = vim.api.nvim_open_win(buf, false, {
+        relative = 'editor',
+        col = 3,
+        row = 3,
+        width = 20,
+        height = 1,
+        style = 'minimal',
+        focusable = false,
+        zindex = 300,
+        cmdline = true,
+        cmdline_offset = -1, -- not inserting "firstc" below
+      })
+      vim.ui_attach(ns, { ext_cmdline = true }, function(event, ...)
+        if event == 'cmdline_show' then
+          local content = select(1, ...)
+          vim.api.nvim_buf_set_lines(buf, -2, -1, false, {content[1][2]})
+          vim.api.nvim_win_set_cursor(_G.win, {1, #content[1][2]})
+          vim.api.nvim__redraw( { win = _G.win, cursor = true, flush = true })
+        end
+        return true
```

Comment:
`:help vim.ui_attach()` doesn't seem to mention the meaning of the return value.

Reply:
Yeah, it makes Nvim not send the event to remote UIs. I'm not sure what the reason behind it is. Here is a related issue: #21150.

Reply:
Indeed (if I understand correctly): the test case uses remote UI and cannot pass without it. By comparison, at TUI it doesn't matter.

Reply:
This test does not test the remote UI state. In this case specifically returning `true` is used to hide the ext_ state from the screen:expect output, the screen class is the remote UI.